integration-tests/offline-test.js

```diff
@@ -1,9 +1,30 @@
 const { expect } = require('chai');
 const visit = require('./helpers/visit');
 
+async function waitForAllServiceWorkers(page) {
+  const allRegistrationsActivated = await page.evaluate(() => navigator.serviceWorker.getRegistrations().then(registrations => {
+    if(!registrations.length) {
+      return false;
+    } 
+
+    let result = registrations.every((registration) => {
+      return registration.active && registration.active.state === 'activated'
+    });
+
+    return result;
+  }));
+  
+  if(!allRegistrationsActivated) {
+    await page.waitFor(500);
+    return waitForAllServiceWorkers(page);
+  }
+}
+
 describe('when offline', function() {
   it('the app loads on the index route', async function() {
-    await visit('/', { waitUntil: 'networkidle0' }, async (page) => {
+    await visit('/', async (page) => {
+      await waitForAllServiceWorkers(page);
+      await page.waitFor(100);
```

Comment:
This indicates that something else is wrong actually. It might not be necessary to do this at all if we switch to `navigator.serviceWorker.ready` maybe but if not I'd like us to look into how we can remove this. We had random waits in the test suite before and ran into problems a few times with that already…